app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,44 @@
+class BookingsController < ApplicationController
+  before_action :set_event
+  before_action :set_booking, only: %i[update cancelation]
+
+  def create
+    @booking = @event.bookings.new(booking_params)
+
+    if @booking.save
+      redirect_to root_path
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
```

Comment:
Where is the event and the user ?

Reply:
the `event` is in the `before_action :set_event` --> `@event = Event.find(params[:event_id])`

the user has been added, thanks for the remark

```
  def create
    @booking = @event.bookings.new(params[:booking])
    @booking.user = current_user

    if @booking.save
      render "pages/home", status: :unprocessable_entity
    end
  end
```

Reply:
Ah you use `@event.bookings.new` instead of `@booking = Booking.new`and `@booking.event = @event` ok
I understand now